packages/flutter_test/lib/src/accessibility.dart

```diff
@@ -438,80 +521,60 @@ class CustomMinimumContrastGuideline extends AccessibilityGuideline {
+    // by HSL lightness and then choose the mode from each group.
+    double totalLightness = 0.0;
+    int count = 0;
+    for (final MapEntry<Color, int> entry in colorHistogram.entries) {
+      totalLightness += HSLColor.fromColor(entry.key).lightness * entry.value;
+      count += entry.value;
     }
-    averageLightness /= colors.length;
+    final double averageLightness = totalLightness / count;
     assert(averageLightness != double.nan);
-    int lightColor = 0;
-    int darkColor = 0;
-    int lightCount = 0;
-    int darkCount = 0;
+
+    MapEntry<Color, int>? lightColor;
+    MapEntry<Color, int>? darkColor;
+
     // Find the most frequently occurring light and dark color.
-    for (final MapEntry<int, int> entry in colorHistogram.entries) {
-      final HSLColor color = HSLColor.fromColor(Color(entry.key));
+    for (final MapEntry<Color, int> entry in colorHistogram.entries) {
+      final double lightness = HSLColor.fromColor(entry.key).lightness;
       final int count = entry.value;
-      if (color.lightness <= averageLightness && count > darkCount) {
-        darkColor = entry.key;
-        darkCount = count;
-      } else if (color.lightness > averageLightness && count > lightCount) {
-        lightColor = entry.key;
-        lightCount = count;
+      if (lightness <= averageLightness) {
+        if (count > (darkColor?.value ?? 0)) {
+          darkColor = entry;
+        }
+      } else {
+        if (count > (lightColor?.value ?? 0)) {
+          lightColor = entry;
+        }
       }
     }
-    // Depending on the number of colors present, return the correct contrast
-    // report.
-    if (lightCount > 0 && darkCount > 0) {
-      return _ContrastReport._(Color(lightColor), Color(darkColor));
-    } else if (lightCount > 0) {
-      return _ContrastReport.singleColor(Color(lightColor));
-    } else if (darkCount > 0) {
-      return _ContrastReport.singleColor(Color(darkColor));
-    } else {
-      return const _ContrastReport.emptyRect();
-    }
-  }
-
-  const _ContrastReport._(this.lightColor, this.darkColor)
-      : isSingleColor = false,
-        isEmptyRect = false;
 
-  const _ContrastReport.singleColor(Color color)
-      : lightColor = color,
-        darkColor = color,
-        isSingleColor = true,
-        isEmptyRect = false;
+    // Text color overrides computed color.
+    // Due to aliasing it can occur less than other shades.
+    if (textColor != null) {
+      if (HSLColor.fromColor(textColor).lightness <= averageLightness) {
+        darkColor = MapEntry<Color, int>(textColor, 1);
+      } else {
+        lightColor = MapEntry<Color, int>(textColor, 1);
+      }
+    }
 
-  const _ContrastReport.emptyRect()
-      : lightColor = _transparent,
-        darkColor = _transparent,
-        isSingleColor = false,
-        isEmptyRect = true;
+    return _ContrastReport._(
+      lightColor?.key ?? darkColor!.key,
+      darkColor?.key ?? lightColor!.key,
```

Comment:
can you help me understand this?

Reply:
If there is only single color, it will be both light and dark.

Reply:
might be worth to note in a comment or add logic that makes this clear

Reply:
Done